examples/python/interceptors/async/async_greeter_server_with_interceptor.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Python AsyncIO implementation of the GRPC helloworld.Greeter server."""
+
+import asyncio
+import contextvars
+import logging
+from typing import Awaitable, Callable, Optional
+
+import grpc
+import helloworld_pb2
+import helloworld_pb2_grpc
+
+rpc_id_var = contextvars.ContextVar('rpc_id', default='default')
+
+
+class ServerInterceptor(grpc.aio.ServerInterceptor):
```

Comment:
Can this be given a more descriptive name?

Reply:
Since we're changing RPC_ID, I'll change it to `RPCIdInterceptor`.